File: algorithm/python/BOJ15650.py
# from https://www.acmicpc.net/problem/15650

import sys
N, M = map(int, sys.stdin.readline().split())


def combination(cnt, start):
    if cnt == M:
        for i in range(0, M):
            print(result_list[i], end=" ")
        print()
        return
    for i in range(start, N):
        result_list[cnt] = input_list[i]
        combination(cnt + 1, i + 1)


input_list = list(map(lambda x: x, range(1, N + 1)))
result_list = [0] * N

combination(0, 0)

File: algorithm/typescript/BOJ15650.ts
// from https://www.acmicpc.net/problem/15650

import fs = require('fs')

let str: string[] = fs.readFileSync('/dev/stdin').toString().split(' ');
let N: number = parseInt(str[0]);
let M: number = parseInt(str[1]);

let input = [];
for (let i = 1; i <= N; i++) {
	input.push(i);
}
let output = [];
let result = [];

function combination(cnt: number, start: number): void {
	if (cnt == M) {
		for (let i = 0; i < M; i++) {
			result.push(`${output[i]} `);
		}
		result.push(`\n`);
		return;
	}

	for (let i = start; i < N; i++) {
		output[cnt] = input[i];
		combination(cnt + 1, i + 1);
	}
}

combination(0, 0);
console.log(result.join(''));
